packages/logic-utils/api.d.ts

```diff
@@ -113,3 +113,18 @@ export function getUI5PropertyByXMLAttributeKey(
   attribute: XMLAttribute,
   model: UI5SemanticModel
 ): UI5Prop | undefined;
+
+/**
+ * Return the UI5 Namespace from the specified or default XML Element namespace
+ *
+ * @param xmlElement
+ * @param model
+ */
+export function getUI5NodeFromXMLElementNamespace(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): {
+  namespace: BaseUI5Node | undefined;
```

Comment:
How can namespace be BaseUI5Node? should it not always be UI5Namespace?
Or is it also possible to return UI5Class (e.g for aggregation tag?)
Basically I am not sure what this utility function does...

Reply:
It returns the UI5 symbol associated with the xml element namespace (or the default namespace if the xml element doesn't have a namespace).
There are classes whose parent isn't a `UI5Namespace` but a different symbol. For `Element` subclasses I could only find 1 such example in version 1.71.* (an enum the contains classes under it), but since it exists, I think we should support it.

Reply:
I would first inspect if that is an API.json error, if it is invalid api.json
I do not mind not supporting 1/100000 edge case.

Even if it is a valid scenario I would prefer to be more specific about that types:
`namespace: UI5Namespace | UI5Enum`

or even:
`namespaceLike: UI5Namespace | UI5Enum`

Reply:
I don't think it's an invalid api.json.
Making the type more specific would make the code more complex because we will have to only search for specific symbol types in the model.
We also have no way to verify which symbol types have sub-symbols in compilation type so we won't know when we have to add types to this `namespaceLike` type.
And while `Element`s are mostly defined on `Namespace`s (except the 3 I found on an enum), there are non-`Element` classes defined on other things like `Class`es.
I think the simple thing is to treat namespaces as being any symbol type. We only use them (as xml namespaces) for resolution of the classes anyway. And if we have other scenarios we can check the `kind`.

Reply:
It might make sense though to define a "concrete" UI5 type (namespace/class/enum etc, things that have a sensible FQN) and use that instead of the `BaseUI5Node` which can be things like properties and aggregations.